treeherder/etl/management/commands/ingest.py

```diff
@@ -154,75 +158,158 @@ def get_decision_task_id(project, revision, root_url):
+                if commiter_date != _commit["commit"]["committer"]["date"]:
+                    event_base_sha = _commit["sha"]
+                    break
+        # This is to make sure that the value has changed
+        assert event_base_sha != repo_meta["branch"]
+        logger.info("We have a new base: %s", event_base_sha)
+        # When using the correct event_base_sha the "commits" field will be correct
+        compareResponse = compare_shas(repo_meta, event_base_sha, commit)
+
     commits = []
-    for _commit in resp["commits"]:
+    for _commit in compareResponse["commits"]:
         commits.append({
             "message": _commit["commit"]["message"],
-            "author": {
-                "name": _commit["commit"]["committer"]["name"],
-                "email": _commit["commit"]["committer"]["email"],
-            },
+            "author": _commit["commit"]["author"],
+            "committer": _commit["commit"]["committer"],
             "id": _commit["sha"],
-            # XXX: A fair estimation
-            "timestamp": _commit["commit"]["author"]["date"]
         })
 
-    pulse = {
+    return event_base_sha, commits
+
+
+def github_push_to_pulse(repo_meta, commit):
+    event_base_sha, commits = query_data(repo_meta, commit)
+
+    return {
         "exchange": "exchange/taskcluster-github/v1/push",
-        "routingKey": "primary.{}.{}".format(owner, repo),
+        "routingKey": "primary.{}.{}".format(repo_meta["owner"], repo_meta["repo"]),
         "payload": {
-            "organization": owner,
+            "organization": repo_meta["owner"],
             "details": {
-                "event.head.repo.url": "https://github.com/{}/{}.git".format(owner, repo),
-                "event.base.repo.branch": branch,
-                # XXX: This is used for the `compare` API. The "event.base.sha" is only contained
-                # in Pulse events, thus, making it hard to correctly establish
-                "event.base.sha": branch,
+                "event.head.repo.url": "{}.git".format(repo_meta["url"]),
+                "event.base.repo.branch": repo_meta["branch"],
+                "event.base.sha": event_base_sha,
                 "event.head.sha": commit,
-                "event.head.user.login": commitInfo["author"]["login"]
             },
             "body": {
                 "commits": commits,
             },
-            "repository": repo
+            "repository": repo_meta["repo"]
         }
     }
-    if merge_base_commit:
-        pulse["payload"]["body"]["merge_base_commit"] = merge_base_commit
-    PushLoader().process(pulse["payload"], pulse["exchange"], root_url)
+
+
+def ingest_git_push(project, commit):
+    _repo = repo_meta(project)
+    pulse = github_push_to_pulse(_repo, commit)
+    PushLoader().process(pulse["payload"], pulse["exchange"], _repo["tc_root_url"])
+
+
+def ingest_git_pushes(project, dry_run=False):
```

Comment:
This method does the work to determine what commits are pushes, ingests them and validates that they are in the same order.